core/src/services/s3/backend.rs

```diff
@@ -932,13 +933,15 @@ impl Access for S3Backend {
                 },
 
                 delete: true,
+                delete_with_versioning: self.core.enable_versioning,
+
                 copy: true,
 
                 list: true,
                 list_with_limit: true,
                 list_with_start_after: true,
                 list_with_recursive: true,
-                list_with_version: self.core.enable_versioning,
+                list_with_versioning: self.core.enable_versioning,
```

Comment:
I suggest we don't change this. Let's continue using `xxx_with_version` until we reach a consensus to add a `versioning` field here.